dev/io.openliberty.org.jboss.resteasy.common/src/org/jboss/resteasy/core/ResourceMethodRegistry.java

```diff
@@ -0,0 +1,523 @@
+
+   /**
+    * Resteasy 2.x does not properly handle sub-resource and sub-resource locator
+    * endpoints with the same uri.  Resteasy 3.x does handle this properly.  In
+    * assisting customers identify this issue during an upgrade from Resteasy 2 to 3
+    * provides a waring when the situation is found.
+    */
+   public void checkAmbiguousUri()
+   {
+      for (Map.Entry<String, List<ResourceInvoker>> entry : this.root.getBounded().entrySet())
+      {
+         List<ResourceInvoker> values = entry.getValue();
+         if (values.size() > 1) {
+            int locatorCnt = 0;
+            int methodCnt = 0;
+            for(ResourceInvoker rInvoker : values)
+            {
+               if (rInvoker instanceof ResourceLocatorInvoker)
+               {
+                  locatorCnt++;
+               } else if (rInvoker instanceof ResourceMethodInvoker)
+               {
+                  methodCnt++;
+               }
+            }
+            if (methodCnt > 0 && locatorCnt > 0)
+            {
+               StringBuilder sb = new StringBuilder();
+               int cnt = values.size();
+               for (int i=0; i < cnt; i++) {
+                  ResourceInvoker exp = values.get(i);
+                  sb.append(exp.getMethod().getDeclaringClass().getName())
+                          .append(".")
+                          .append(exp.getMethod().getName());
+                  if (i < cnt-1)
+                  {
+                     sb.append(", ");
+                  }
+               }
+               LogMessages.LOGGER.uriAmbiguity(entry.getKey(), sb.toString());
+            }
+         }
+      }
+   }
+
+   protected void processMethod(ResourceFactory rf, String base, ResourceLocator method)
+   {
+      ResteasyUriBuilder builder = new ResteasyUriBuilderImpl();
+      if (base != null)
+         builder.path(base);
+      builder.path(method.getFullpath());
+      String fullpath = builder.getPath();
+      if (fullpath == null)
+         fullpath = "";
+
+      // Liberty Change Start
+      String httpMethod = "unknown";
+      if (method.getMethod().getAnnotation(GET.class) != null) {
+          httpMethod = "GET";
+      } else if (method.getMethod().getAnnotation(POST.class) != null) {
+          httpMethod = "POST";
+      } else if (method.getMethod().getAnnotation(PUT.class) != null) {
+          httpMethod = "PUT";
+      } else if (method.getMethod().getAnnotation(DELETE.class) != null) {
+          httpMethod = "DELETE";
+      }
+          
+      ServletContext sc = (ServletContext) ResteasyContext.getContextDataMap().get(ServletContext.class);
+      if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+          Method m = method.getMethod();
+          Class parent = m.getDeclaringClass();
+          String parentName = parent.getName();
+          String message = "- " + httpMethod + " " /*+ sc.getContextPath()*/ + fullpath + " - " + parentName + "#" + method.getMethod().getName();
```

Comment:
Do we want to print the context root with each endpoint, or is printing the app name once enough? I kinda like printing the FULL path.

Reply:
I like the full path.